src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -5621,13 +5621,16 @@ internal void SetReceivingPacketInformation()
                 IPEndPoint ipEndPoint = _rightEndPoint as IPEndPoint;
                 IPAddress boundAddress = (ipEndPoint != null ? ipEndPoint.Address : null);
                 Debug.Assert(boundAddress != null, "Not Bound");
-                if (_addressFamily == AddressFamily.InterNetwork
-                    || (boundAddress != null && IsDualMode
-                        && (boundAddress.IsIPv4MappedToIPv6 || boundAddress.Equals(IPAddress.IPv6Any))))
+                if (_addressFamily == AddressFamily.InterNetwork)
                 {
                     SetSocketOption(SocketOptionLevel.IP, SocketOptionName.PacketInformation, true);
                 }
 
+                if ((boundAddress != null && IsDualMode && (boundAddress.IsIPv4MappedToIPv6 || boundAddress.Equals(IPAddress.IPv6Any))))
```

Comment:
Maybe `else if` instead of `if`?
We shouldn't call setsockopt twice with the same configuration.